src/controllers/session.ts

```diff
@@ -112,67 +116,95 @@ const deleteSession = async (req: Request, res: Response) => {
 };
 
 const updateStatus = async (req: Request, res: Response) => {
-  const { userStatus } = req.body;
   const { sessionId } = req.params;
+  const userId = req.user.userId;
+  const { status } = req.body;
+
+  const session = await SessionModel.findById({ _id: sessionId });
+
+  if (!session) {
+    return res.status(404).json({ message: 'Session not found' });
+  }
+
+  const user = await User.findById({ _id: userId });
 
-  const sessionUser = await SessionModel.findOneAndUpdate(
-    { _id: sessionId, 'participant.user.userInfo': req.user.userId },
-    { $set: { 'participant.$.user.userStatus': userStatus } },
-    { new: true }
+  if (!user) {
+    return res.status(404).json({ message: 'User not found' });
+  }
+
+  const participantIds = session.participant.map((participant) =>
+    (participant as any)._id.toString()
   );
-  if (!sessionUser) {
-    const newSessionUser = await SessionModel.findOneAndUpdate(
-      { _id: sessionId },
-      {
-        $addToSet: {
-          participant: {
-            user: {
-              userInfo: req.user.userId,
-              userStatus,
-            },
-          },
-        },
-      },
-      { new: true }
+
+  let event;
```

Comment:
This variable is not being used anywhere, we could also remove the commented code.